### DISCUSSION TITLE
How to load multiple models into a viewport and support viewing the properties of each model

### DISCUSSION THREAD

**Discussion by xihumeijing** [iTwin/itwinjs-core#5814] [Q&A]
I tried to implement it using the following method, but the added model does not support viewing properties.

```
const vp = IModelApp.viewManager.selectedView;
const iModel = await BriefcaseConnection.openFile({readonly: true, fileName: filename });
const provider = await Provider.create(vp, iModel);
vp.addTiledGraphicsProvider(provider);
vp.invalidateScene();
```

**Comment by pmconne** [comment]
In its current form, a Viewport always operates on one iModel (`Viewport.iModel`). It doesn't know about other iModels you may have added as `TiledGraphicsProvider`s. Nearly all tools and UI widgets assume that they are operating on `Viewport.iModel`.

We are in the process of evaluating how we can relax this restriction to permit easily visualizing any number of iModels within a viewport. But until then, you'll have to write your own selection tool that can identify which iModel the user is requesting properties for and route the request appropriately. Or, you might consider combining your iModels into one iModel.

**Reply by xihumeijing** [reply]
How to combining two iModels into one iModel? use **[imodel-transformer](https://github.com/iTwin/imodel-transformer)**?

**Reply by pmconne** [reply]
Yes, you can use that package to combine the iModels yourself, or you can use the [combine iModels](https://developer.bentley.com/apis/transformations/operations/combineimodels/) REST API.
Note, the imodel-transformer package is currently in **pre-release** (version 0.3.0), meaning its API is not guaranteed to remain stable.
